publication-doi-commons/src/main/java/no/unit/nva/publication/doi/PublicationMapper.java

```diff
@@ -34,12 +44,54 @@ private static URI transformIdentifierToId(String namespace, String identifier)
     }
 
     /**
-     * Map to doi.{@link Publication} from {@link DynamodbStreamRecordDao}.
+     * Map a DynamodbStreamRecord with oldImage and/or newImage to PublicationMapping. Publication is a wrapper object
+     * containing mapped old and/or new Publication.
      *
-     * @param dao {@link DynamodbStreamRecordDao}
+     * @param streamRecord  DynamodbStreamRecord
+     * @return PublicationMapping
+     */
+    public PublicationMapping fromDynamodbStreamRecord(DynamodbStreamRecord streamRecord) {
+        var publicationMappingBuilder = PublicationMapping.Builder.newBuilder();
+        var dynamodb = streamRecord.getDynamodb();
+
+        if (dynamodb != null) {
+            var streamViewType = dynamodb.getStreamViewType();
```

Comment:
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_StreamSpecification.html

What about the other cases?

Reply:
Maybe even throw RuntimeException of unknowns to detect when we get the edge cases? And if we have to deal with em?